internal/informer/informer.go

```
@@ -120,6 +120,14 @@ func (w *DeploymentWatcher) Run(ctx context.Context) error {
 	w.factory.Start(ctx.Done())
 
 	if err := w.WaitForSync(ctx); err != nil {
+		// Cancellation during the initial sync is a shutdown, not a failure:
+		// a SIGTERM that lands while the watch is still establishing must end
+		// the process cleanly, the same as one that lands an hour later.
+		if ctx.Err() != nil {
+			w.factory.Shutdown()
+			w.logger.Info().Msg("Cancelled during initial sync, deployment informer stopped")
+			return nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 126) but it returns nil

(nilerr)